BUILD

```diff
@@ -13,28 +19,87 @@ cc_library(
     ],
 )
 
-# TODO(mathetkae): once other runtimes(WAVM,V8) can be linked in this repos,
-#   use -define=wasm=v8|wavm|wasmtime and switch
 cc_library(
-    name = "lib",
-    srcs = glob(
-        [
-            "src/**/*.cc",
-            "src/**/*.h",
-        ],
-        exclude = [
-            "src/**/v8*",
-            "src/**/wamr*",
-            "src/**/wavm*",
-        ],
-    ),
-    hdrs = glob(["src/**/*.h"]),
-    copts = COPTS,
+    name = "common_lib",
+    srcs = glob([
+        "src/*.h",
+        "src/*.cc",
+        "src/common/*.h",
+        "src/common/*.cc",
+        "src/third_party/*.h",
+        "src/third_party/*.cc",
+        "src/null/*.cc",
+    ]),
     deps = [
         ":include",
         "@boringssl//:crypto",
         "@com_google_protobuf//:protobuf_lite",
         "@proxy_wasm_cpp_sdk//:api_lib",
+    ],
+)
+
+cc_library(
+    name = "v8_lib",
+    srcs = glob([
+        # TODO(@mathetake): Add V8 lib.
+        # "src/v8/*.h",
+        # "src/v8/*.cc",
+    ]),
+    deps = [
+        ":common_lib",
+        # TODO(@mathetake): Add V8 lib.
+    ],
+)
+
+cc_library(
+    name = "wamr_lib",
+    srcs = glob([
+        # TODO(@mathetake): Add WAMR lib.
+        # "src/wamr/*.h",
+        # "src/wamr/*.cc",
+    ]),
+    deps = [
+        ":common_lib",
+        # TODO(@mathetake): Add WAMR lib.
+    ],
+)
+
+cc_library(
+    name = "wasmtime_lib",
+    srcs = glob([
+        "src/wasmtime/*.h",
+        "src/wasmtime/*.cc",
+    ]),
+    deps = [
+        ":common_lib",
         "@wasm_c_api//:wasmtime_lib",
     ],
 )
+
+cc_library(
+    name = "wavm_lib",
+    srcs = glob([
+        # TODO(@mathetake): Add WAVM lib.
+        # "src/wavm/*.h",
+        # "src/wavm/*.cc",
+    ]),
+    deps = [
+        ":common_lib",
+        # TODO(@mathetake): Add WAVM lib.
```

Comment:
Any chance you could try to add WAVM here? Or do you prefer to leave it for a separate PR?

Reply:
I will do in a separate PR.